jax/_src/dispatch.py

```diff
@@ -571,6 +572,15 @@ def jaxpr_shardings(jaxpr) -> Iterator[jax.sharding.XLACompatibleSharding]:
     elif eqn.primitive is pjit.pjit_p:
       yield from eqn.params['in_shardings']
       yield from eqn.params['out_shardings']
+    # TODO(mattjj): replace string matching with object id
```

Comment:
This will directly affect pjit i.e. if you have a shard_map inside pjit, the NamedSharding will be taken into account for getting the device assignment. That is intended right?

Reply:
Yes! That's the point of the change.

I can fix the TODO now though, thanks for the reminder...

Reply:
Done!